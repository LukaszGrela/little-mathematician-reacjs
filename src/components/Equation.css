.equation {
    margin: 0 auto;
    width: 300px;
}

.part {
    width: 50px;
    padding: 15px 5px;
    margin: 5px;
    display: table-cell;
}
.question {
    border: 1px solid #ccc;
}