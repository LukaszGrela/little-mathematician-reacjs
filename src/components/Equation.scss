/*
   Copyright 2018 ≈Åukasz 'Severiaan' Grela GrelaDesign

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

@import '../sass/_variables.scss';
.equation {
    margin: 0 auto;
    padding: 75px 0;
    width: $width-game-column;
    display: table;
}
.wrapper {
    display: table-cell;
    text-align: center;
    vertical-align: middle;
}

.part {
    width: 50px;
    padding: 15px 5px;
    margin: 0px;
    display: inline-block;
    vertical-align: top;
}

.question {
    border: 2px solid $divider-color;
    border-radius: 5px;
    .correct & {
        border-color: $color-correct;
        border-radius: 15px;
    }
    .incorrect & {
        border-color: $color-incorrect;
        border-radius: 0;
    }
}
