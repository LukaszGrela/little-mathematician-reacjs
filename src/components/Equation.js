/*
   Copyright 2018 ≈Åukasz 'Severiaan' Grela GrelaDesign

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/
import React, { Component } from 'react';

import './Equation.scss'

class Equation extends Component {
    constructor(props) {
        super(props);
        this.state = {};
        console.log('Equation#constructor', props);
    }
    render() {
        const { operandA, operandB, result, ask, operation, answer } = this.props;
        let userAnswer = answer ? answer.user : '?';
        return (
            <div className={"equation " + (
                answer ? (
                    answer.correct ? 'correct' : 'incorrect'
                ) : ''
            )}>
                <div className={'wrapper'}>
                    <div className={"part operand-a" + (ask === 'a' ? ' question' : '')}>{ask === 'a' ? userAnswer : operandA}</div>
                    <div className={"part operation"}>{operation}</div>
                    <div className={"part operand-b" + (ask === 'b' ? ' question' : '')}>{ask === 'b' ? userAnswer : operandB}</div>
                    <div className={"part equation"}>=</div>
                    <div className={"part result" + (ask === 'result' ? ' question' : '')}>{ask === 'result' ? userAnswer : result}</div>
                </div>
            </div>
        );
    }
}

export default Equation;
