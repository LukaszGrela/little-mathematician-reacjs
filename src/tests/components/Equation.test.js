import React from 'react';
import { shallow } from 'enzyme';
import Equation from '../../components/Equation';

let wrapper, operation, userAnswer,
    questions = {
        id: 0,
        operandA: 1,
        operandB: 2,
        result: 3,
        ask: "result",
        correct: 3,
        distractors: [1, 2, 3, 4],
        operation: '+',
    };
beforeEach(() => {
    operation = 'A';
    userAnswer = { correct: true };
    wrapper = shallow(<Equation
        {...questions} />);
})

test('Should render Equation correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Should Equation set operation correctly', () => {
    wrapper.setProps({ operation: '-' });
    let div = wrapper.find('.operation');
    expect(div).toHaveLength(1);
    expect(div.text()).toBe("-");

    wrapper.setProps({ operation: '+' });
    div = wrapper.find('.operation');
    expect(div).toHaveLength(1);
    expect(div.text()).toBe("+");
});

test('Should Equation put "?" in the operand A', () => {
    wrapper.setProps({ ask: 'a' });
    const div = wrapper.find('.operand-a.question');
    expect(div).toHaveLength(1);

    expect(div.text()).toBe("?");
});
test('Should Equation put "userAnswer" value in the operand A', () => {
    const user = 10;
    wrapper.setProps({ ask: 'a', answer: { user } });
    const div = wrapper.find('.operand-a.question');
    expect(div).toHaveLength(1);

    expect(div.text()).toBe("" + user);
});

test('Should Equation put "?" in the operand B', () => {
    wrapper.setProps({ ask: 'b' });
    const div = wrapper.find('.operand-b.question');
    expect(div).toHaveLength(1);

    expect(div.text()).toBe("?");
});
test('Should Equation put "userAnswer" value in the operand B', () => {
    const user = 10;
    wrapper.setProps({ ask: 'b', answer: { user } });
    const div = wrapper.find('.operand-b.question');
    expect(div).toHaveLength(1);

    expect(div.text()).toBe("" + user);
});

test('Should Equation put "?" in the result', () => {

    const div = wrapper.find('.result.question');
    expect(div).toHaveLength(1);

    expect(div.text()).toBe("?");
});
test('Should Equation put "userAnswer" value in the result', () => {
    const user = 10;
    wrapper.setProps({ answer: { user } });
    const div = wrapper.find('.result.question');
    expect(div).toHaveLength(1);

    expect(div.text()).toBe("" + user);
});

test('Should Equation className NOT have correct or incorrect class', () => {

    let div = wrapper.find('.equation.correct');
    expect(div).toHaveLength(0);
    
    div = wrapper.find('.equation.incorrect');
    expect(div).toHaveLength(0);
    
});
test('Should Equation className have proper correct/incorrect class', () => {

    wrapper.setProps({ answer: { correct:true } });
    let div = wrapper.find('.equation.correct');
    expect(div).toHaveLength(1);
    
    wrapper.setProps({ answer: { correct:false } });
    div = wrapper.find('.equation.incorrect');
    expect(div).toHaveLength(1);
    
});