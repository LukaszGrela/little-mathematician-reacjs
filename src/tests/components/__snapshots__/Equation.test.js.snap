// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Equation correctly 1`] = `
<div
  className="equation "
>
  <div
    className="wrapper"
  >
    <div
      className="part operand-a"
    >
      1
    </div>
    <div
      className="part operation"
    >
      +
    </div>
    <div
      className="part operand-b"
    >
      2
    </div>
    <div
      className="part equation"
    >
      =
    </div>
    <div
      className="part result question"
    >
      ?
    </div>
  </div>
</div>
`;
